src/components/Api.js

```diff
@@ -3,19 +3,44 @@ export default class Api {
     this._baseUrl = baseUrl;
     this._headers = headers;
   }
-  getAppInformation() {
-    return Promise.all([this.getInitialCards(), this.getUserInformation()]);
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
   }
 
-  getInitialCards() {
-    return fetch(`${this._baseUrl}/cards`, { headers: this._headers })
-      .then((res) => {
-        if (res.ok) {
-          return res.json();
-        }
-        // if the server returns an error, reject the promise
-        return Promise.reject(`Error: ${res.status}`);
+  _request(url, options) {
+    return fetch(url, options).then(this._checkResponse);
+  }
+
+  getUserInformation() {
+    return this._request(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    }).finally(() => {
+      console.log("Done with getting user info");
+    });
+  }
+
+  updateProfilePicture({ avatar }) {
+    return this._request(`${this._baseUrl}/users/me/avatar`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        avatar,
+      }),
+    })
+      .then((result) => {
+        return result;
       })
+      .finally(() => {
+        console.log("Done running Avatar change API");
+      });
+  }
+
+  getInitialCards() {
+    return this._request(`${this._baseUrl}/cards`, { headers: this._headers })
       .then((result) => {
         return result;
       })
```

Comment:
Please, delete `catch` from line 47

Reply:
I've made and pushed the changes. I'm unsure what you mean by the comment "userProfileAvatar.src = response.avatar; Class UserInfo should set the entire user’s data (including the avatar)"
I currently do have that functionality under the setUserInfo() method inside UserInfo class
If that is supposed to be elsewhere, where should I put it?

Reply:
> "userProfileAvatar.src = response.avatar;

I have some problems with my screen shooter and you didn't have this code in `files changed`


I meant that you set the avatar not with `setUserInfo` in line 125  of `index.js` you have  "userProfileAvatar.src = response.avatar; but you need to set it correctly